lib/storage/metadata/mongoclient/MongoClientInterface.ts

```diff
@@ -2708,13 +3232,17 @@ class MongoClientInterface {
      * @param {Function} cb callback
      * @return {undefined}
      */
-    deleteBucketIndexes(bucketName, indexSpecs, log, cb) {
-        const c = this.getCollection(bucketName);
+    deleteBucketIndexes(bucketName: string, indexSpecs: { name: string }[], log: werelogs.Logger, cb: NodeCallBack<void>) {
+        const c = this.getCollection<ObjectMetastoreDocument>(bucketName);
+        if (!c) {
+            log.error('deleteBucketIndexes: error getting bucket collection', { bucketName });
+            return cb(errors.InternalError);
+        }
         async.each(indexSpecs,
             (spec, next) => c.dropIndex(spec.name).then(() => next()).catch(err => next(err)),
             err => {
                 if (err) {
-                    if (err.codeName === 'NamespaceNotFound') {
+                    if (err instanceof MongoServerError && err.codeName === 'NamespaceNotFound') {
```

Comment:
do we need the instance test?
can we do something like `err.codeName? === 'NamespaceNotFound'`, since we don't really care about the type (only that it may have a codeName field, and in that case, if the value match)?